sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ParametersCorrectOrder.cs

```diff
@@ -117,15 +117,14 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                 var parameterName = parameter.Name;
 
                 if (string.IsNullOrEmpty(identifierName) ||
-                    !parameterNames.Contains(identifierName))
+                    !parameterNames.Contains(identifierName) ||
+                    !IdentifierWithSameNameAndTypeExists(parameter))

```

Comment:
We are excluding the parameters that do not have a corresponding variable with the same name and type (including parameters declared in the calling method, fields, etc.).